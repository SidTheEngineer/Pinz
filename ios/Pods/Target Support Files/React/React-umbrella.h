#ifdef __OBJC__
#import <UIKit/UIKit.h>
#endif

#import "RCTAssert.h"
#import "RCTBridge+Private.h"
#import "RCTBridge.h"
#import "RCTBridgeDelegate.h"
#import "RCTBridgeMethod.h"
#import "RCTBridgeModule.h"
#import "RCTBundleURLProvider.h"
#import "RCTConvert.h"
#import "RCTDefines.h"
#import "RCTDisplayLink.h"
#import "RCTErrorCustomizer.h"
#import "RCTErrorInfo.h"
#import "RCTEventDispatcher.h"
#import "RCTFrameUpdate.h"
#import "RCTImageSource.h"
#import "RCTInvalidating.h"
#import "RCTJavaScriptExecutor.h"
#import "RCTJavaScriptLoader.h"
#import "RCTJSStackFrame.h"
#import "RCTKeyCommands.h"
#import "RCTLog.h"
#import "RCTModuleData.h"
#import "RCTModuleMethod.h"
#import "RCTMultipartDataTask.h"
#import "RCTMultipartStreamReader.h"
#import "RCTNullability.h"
#import "RCTParserUtils.h"
#import "RCTPerformanceLogger.h"
#import "RCTRootView.h"
#import "RCTRootViewDelegate.h"
#import "RCTRootViewInternal.h"
#import "RCTTouchEvent.h"
#import "RCTTouchHandler.h"
#import "RCTURLRequestDelegate.h"
#import "RCTURLRequestHandler.h"
#import "RCTUtils.h"
#import "RCTWebSocketProxy.h"
#import "RCTWebSocketProxyDelegate.h"
#import "CSSLayout.h"
#import "CSSMacros.h"
#import "CSSNodeList.h"
#import "RCTJSCErrorHandling.h"
#import "RCTJSCExecutor.h"
#import "RCTJSCWrapper.h"
#import "RCTAccessibilityManager.h"
#import "RCTAlertManager.h"
#import "RCTAppState.h"
#import "RCTAsyncLocalStorage.h"
#import "RCTClipboard.h"
#import "RCTDevLoadingView.h"
#import "RCTDevMenu.h"
#import "RCTEventEmitter.h"
#import "RCTExceptionsManager.h"
#import "RCTI18nManager.h"
#import "RCTI18nUtil.h"
#import "RCTKeyboardObserver.h"
#import "RCTRedBox.h"
#import "RCTSourceCode.h"
#import "RCTStatusBarManager.h"
#import "RCTTiming.h"
#import "RCTUIManager.h"
#import "RCTFPSGraph.h"
#import "RCTJSCProfiler.h"
#import "RCTMacros.h"
#import "RCTProfile.h"
#import "RCTActivityIndicatorView.h"
#import "RCTActivityIndicatorViewManager.h"
#import "RCTAnimationType.h"
#import "RCTAutoInsetsProtocol.h"
#import "RCTBorderDrawing.h"
#import "RCTBorderStyle.h"
#import "RCTComponent.h"
#import "RCTComponentData.h"
#import "RCTConvert+CoreLocation.h"
#import "RCTConvert+MapKit.h"
#import "RCTDatePicker.h"
#import "RCTDatePickerManager.h"
#import "RCTFont.h"
#import "RCTMap.h"
#import "RCTMapAnnotation.h"
#import "RCTMapManager.h"
#import "RCTMapOverlay.h"
#import "RCTModalHostView.h"
#import "RCTModalHostViewController.h"
#import "RCTModalHostViewManager.h"
#import "RCTNavigator.h"
#import "RCTNavigatorManager.h"
#import "RCTNavItem.h"
#import "RCTNavItemManager.h"
#import "RCTPicker.h"
#import "RCTPickerManager.h"
#import "RCTPointerEvents.h"
#import "RCTProgressViewManager.h"
#import "RCTRefreshControl.h"
#import "RCTRefreshControlManager.h"
#import "RCTRootShadowView.h"
#import "RCTScrollableProtocol.h"
#import "RCTScrollView.h"
#import "RCTScrollViewManager.h"
#import "RCTSegmentedControl.h"
#import "RCTSegmentedControlManager.h"
#import "RCTShadowView.h"
#import "RCTSlider.h"
#import "RCTSliderManager.h"
#import "RCTSwitch.h"
#import "RCTSwitchManager.h"
#import "RCTTabBar.h"
#import "RCTTabBarItem.h"
#import "RCTTabBarItemManager.h"
#import "RCTTabBarManager.h"
#import "RCTTextDecorationLineType.h"
#import "RCTView.h"
#import "RCTViewControllerProtocol.h"
#import "RCTViewManager.h"
#import "RCTWebView.h"
#import "RCTWebViewManager.h"
#import "RCTWrapperViewController.h"
#import "UIView+Private.h"
#import "UIView+React.h"
#import "RCTActionSheetManager.h"
#import "RCTLocationObserver.h"
#import "RCTGIFImageDecoder.h"
#import "RCTImageBlurUtils.h"
#import "RCTImageCache.h"
#import "RCTImageEditingManager.h"
#import "RCTImageLoader.h"
#import "RCTImageStoreManager.h"
#import "RCTImageUtils.h"
#import "RCTImageView.h"
#import "RCTImageViewManager.h"
#import "RCTLocalAssetImageLoader.h"
#import "RCTResizeMode.h"
#import "RCTLinkingManager.h"
#import "RCTDataRequestHandler.h"
#import "RCTFileRequestHandler.h"
#import "RCTHTTPRequestHandler.h"
#import "RCTNetInfo.h"
#import "RCTNetworking.h"
#import "RCTNetworkTask.h"
#import "RCTSettingsManager.h"
#import "RCTRawTextManager.h"
#import "RCTShadowRawText.h"
#import "RCTShadowText.h"
#import "RCTText.h"
#import "RCTTextField.h"
#import "RCTTextFieldManager.h"
#import "RCTTextManager.h"
#import "RCTTextSelection.h"
#import "RCTTextView.h"
#import "RCTTextViewManager.h"
#import "RCTVibration.h"
#import "RCTSRWebSocket.h"
#import "RCTWebSocketExecutor.h"
#import "RCTWebSocketManager.h"
#import "RCTWebSocketModule.h"

FOUNDATION_EXPORT double ReactVersionNumber;
FOUNDATION_EXPORT const unsigned char ReactVersionString[];

