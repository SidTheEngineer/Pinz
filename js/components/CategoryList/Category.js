import React from 'react';
import {
  View,
  Text,
  TouchableOpacity,
  StyleSheet
} from 'react-native';
import Icon from 'react-native-vector-icons/FontAwesome';

const styles = StyleSheet.create({
  categoryContainer: {
    height: 75,
    flexDirection: 'row',
    borderColor: 'rgb(150,150,150)',
    borderStyle: 'solid',
    borderTopWidth: 0.5,
    borderBottomWidth: 0.5,
  },
  textContainer: {
    flex: 1,
    margin: 5,
    flexDirection: 'row',
    alignItems: 'center',
    justifyContent: 'space-between',
  }
});

const Category = props => (
  <TouchableOpacity style={styles.categoryContainer} onPress={props.navigate}>
    <View style={styles.textContainer}>
      <Text>{ props.title }</Text>
      <Icon name="arrow-right" size={20} style={{ marginRight: 15 }} />
    </View>
  </TouchableOpacity>
);

Category.propTypes = {
  navigate: React.PropTypes.func,
  title: React.PropTypes.string.isRequired
};

export default Category;
