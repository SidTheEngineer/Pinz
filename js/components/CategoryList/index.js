import React, { Component } from 'react';
import {
  ScrollView,
  StyleSheet,
  BackAndroid
} from 'react-native';

import { COLORS } from '../../constant';
import Category from './Category';

const styles = StyleSheet.create({
  categoryListContainer: {
    flex: 1,
    backgroundColor: COLORS.LIGHT_GRAY,
    marginTop: 54
  }
});

class CategoryList extends Component {

  constructor() {
    super();

    this.navigate = this.navigate.bind(this);
  }

  navigate(title) {
    this.props.navigator.push({
      title
    });
  }

  render() {
    const { navigator } = this.props;

    BackAndroid.addEventListener('hardwareBackPress', () => {
      if (navigator.getCurrentRoutes().length > 1) {
        navigator.pop();
        return true;
      }
      return false;
    });

    return (
      <ScrollView style={styles.categoryListContainer}>
        <Category {...this.props} title={'Posts'} navigate={() => this.navigate('Posts')} />
        <Category {...this.props} title={'Events'} />
        <Category {...this.props} title={'Garages'} />
        <Category {...this.props} title={'Food'} />
        <Category {...this.props} title={'Test'} />
      </ScrollView>
    );
  }
}

CategoryList.propTypes = {
  navigator: React.PropTypes.shape({
    push: React.PropTypes.func.isRequired,
    pop: React.PropTypes.func.isRequired,
    getCurrentRoutes: React.PropTypes.func.isRequired
  }),
};

export default CategoryList;
