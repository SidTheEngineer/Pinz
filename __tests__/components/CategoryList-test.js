import 'react-native';
import React from 'react';
import renderer from 'react-test-renderer';

import CategoryList from '../../js/components/CategoryList';

describe('CategoryList', () => {
  it('renders correctly', () => {
    const categoryList = renderer.create(
      <CategoryList />
    ).toJSON();
    expect(categoryList).toMatchSnapshot();
  });
});
