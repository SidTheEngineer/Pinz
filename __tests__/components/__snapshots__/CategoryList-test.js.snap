exports[`CategoryList renders correctly 1`] = `
<ScrollView
  style={
    Object {
      "backgroundColor": "rgb(235,235,235)",
      "flex": 1,
      "marginTop": 64,
    }
  }>
  <TouchableOpacity
    activeOpacity={0.2}
    onPress={[Function]}
    style={
      Object {
        "flexDirection": "row",
        "height": 75,
      }
    }>
    <View
      style={
        Object {
          "alignItems": "center",
          "flex": 1,
          "flexDirection": "row",
          "justifyContent": "space-between",
          "margin": 5,
        }
      }>
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Object {
            "color": "rgb(100,100,100)",
            "fontWeight": "bold",
            "marginLeft": 15,
          }
        }>
        Events
      </Text>
      <Text
        accessible={true}
        allowFontScaling={false}
        ellipsizeMode="tail"
        style={
          Array [
            Object {
              "color": "rgb(100,100,100)",
              "fontSize": 20,
            },
            Object {
              "marginRight": 15,
            },
            Object {
              "fontFamily": "FontAwesome",
              "fontStyle": "normal",
              "fontWeight": "normal",
            },
          ]
        }>
        
      </Text>
    </View>
  </TouchableOpacity>
  <TouchableOpacity
    activeOpacity={0.2}
    onPress={undefined}
    style={
      Object {
        "flexDirection": "row",
        "height": 75,
      }
    }>
    <View
      style={
        Object {
          "alignItems": "center",
          "flex": 1,
          "flexDirection": "row",
          "justifyContent": "space-between",
          "margin": 5,
        }
      }>
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Object {
            "color": "rgb(100,100,100)",
            "fontWeight": "bold",
            "marginLeft": 15,
          }
        }>
        Garages
      </Text>
      <Text
        accessible={true}
        allowFontScaling={false}
        ellipsizeMode="tail"
        style={
          Array [
            Object {
              "color": "rgb(100,100,100)",
              "fontSize": 20,
            },
            Object {
              "marginRight": 15,
            },
            Object {
              "fontFamily": "FontAwesome",
              "fontStyle": "normal",
              "fontWeight": "normal",
            },
          ]
        }>
        
      </Text>
    </View>
  </TouchableOpacity>
  <TouchableOpacity
    activeOpacity={0.2}
    onPress={undefined}
    style={
      Object {
        "flexDirection": "row",
        "height": 75,
      }
    }>
    <View
      style={
        Object {
          "alignItems": "center",
          "flex": 1,
          "flexDirection": "row",
          "justifyContent": "space-between",
          "margin": 5,
        }
      }>
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Object {
            "color": "rgb(100,100,100)",
            "fontWeight": "bold",
            "marginLeft": 15,
          }
        }>
        Food
      </Text>
      <Text
        accessible={true}
        allowFontScaling={false}
        ellipsizeMode="tail"
        style={
          Array [
            Object {
              "color": "rgb(100,100,100)",
              "fontSize": 20,
            },
            Object {
              "marginRight": 15,
            },
            Object {
              "fontFamily": "FontAwesome",
              "fontStyle": "normal",
              "fontWeight": "normal",
            },
          ]
        }>
        
      </Text>
    </View>
  </TouchableOpacity>
  <TouchableOpacity
    activeOpacity={0.2}
    onPress={undefined}
    style={
      Object {
        "flexDirection": "row",
        "height": 75,
      }
    }>
    <View
      style={
        Object {
          "alignItems": "center",
          "flex": 1,
          "flexDirection": "row",
          "justifyContent": "space-between",
          "margin": 5,
        }
      }>
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Object {
            "color": "rgb(100,100,100)",
            "fontWeight": "bold",
            "marginLeft": 15,
          }
        }>
        Test
      </Text>
      <Text
        accessible={true}
        allowFontScaling={false}
        ellipsizeMode="tail"
        style={
          Array [
            Object {
              "color": "rgb(100,100,100)",
              "fontSize": 20,
            },
            Object {
              "marginRight": 15,
            },
            Object {
              "fontFamily": "FontAwesome",
              "fontStyle": "normal",
              "fontWeight": "normal",
            },
          ]
        }>
        
      </Text>
    </View>
  </TouchableOpacity>
</ScrollView>
`;
